UPGRADE-4.2.md

```diff
@@ -161,8 +181,66 @@ FrameworkBundle
 Messenger
 ---------
 
- * `EnvelopeItemInterface` doesn't extend `Serializable` anymore
- * The `handle` method of the `Symfony\Component\Messenger\Middleware\ValidationMiddleware` and `Symfony\Component\Messenger\Asynchronous\Middleware\SendMessageMiddleware` middlewares now requires an `Envelope` object to be given (because they implement the `EnvelopeAwareInterface`). When using these middleware with the provided `MessageBus`, you will not have to do anything. If you use the middlewares any other way, you can use `Envelope::wrap($message)` to create an envelope for your message.
+ * The `MiddlewareInterface::handle()` and `SenderInterface::send()` methods must now return an `Envelope` instance.
+ * Handlers must not return any value (`void`). If you used to return a value, e.g in query bus handlers, you can either:
+    - make your `Query` mutable to allow setting & getting a result:
+      ```php
+      // When dispatching:
+      $bus->dispatch($query = new Query());
+      $result = $query->getResult();
+
+      // In your handler:
+      $query->setResult($yourResult);
+      ```
+    - define a callable on your `Query` to be called in your handler:
+      ```php
+      // When dispatching:
+      $bus->dispatch(new Query([$this, 'onResult']));
+
+      // In your handler:
+      $query->executeCallback($yourResult);
+      ```
+
+ * The `EnvelopeAwareInterface` was removed and the `MiddlewareInterface::handle()` method now requires an `Envelope` object
+   as first argument. When using built-in middleware with the provided `MessageBus`, you will not have to do anything.  
+   If you use your own `MessageBusInterface` implementation, you can use `Envelope::wrap($message)` to create an envelope for your messages.  
```

Comment:
`Envelope::wrap()` doesn't exist anymore

Reply:
Right. In my mind it was re-introduced in another PR but was closed instead 😅